File: app/src/main/java/com/example/yollotl/AlmanaqueActivity.java
/* package com.example.yollotl;

import android.os.Bundle;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class AlmanaqueActivity extends AppCompatActivity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_almanaque);

        RecyclerView recyclerView = findViewById(R.id.recycler_view_plantas);
        recyclerView.setLayoutManager(new LinearLayoutManager(this));

        PlantaApi plantaApi = RetrofitClient.getInstance().create(PlantaApi.class);
        plantaApi.getPlantas().enqueue(new Callback<List<Planta>>() {
            @Override
            public void onResponse(Call<List<Planta>> call, Response<List<Planta>> response) {
                if (response.isSuccessful() && response.body() != null) {
                    PlantaAdapter adapter = new PlantaAdapter(response.body());
                    recyclerView.setAdapter(adapter);
                }
            }

            @Override
            public void onFailure(Call<List<Planta>> call, Throwable t) {
                Toast.makeText(AlmanaqueActivity.this, "Error al cargar datos", Toast.LENGTH_SHORT).show();
            }
        });
    }
}
*/

File: app/src/main/java/com/example/yollotl/AlmanaqueActivity.kt
package com.example.yollotl

import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class AlmanaqueActivity : AppCompatActivity() {

    private lateinit var plantaAdapter: PlantaAdapter
    private lateinit var recyclerView: RecyclerView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_almanaque)

        recyclerView = findViewById(R.id.recyclerView)
        recyclerView.layoutManager = LinearLayoutManager(this)
        plantaAdapter = PlantaAdapter(emptyList())
        recyclerView.adapter = plantaAdapter

        // Debug log to check initialization
        Log.d("AlmanaqueActivity", "RecyclerView and Adapter initialized")

        fetchPlantas()
    }

    private fun fetchPlantas() {
        // Debug log to check fetching process
        Log.d("AlmanaqueActivity", "Fetching plantas")

        RetrofitClient.instance.getPlantas().enqueue(object : Callback<List<Planta>> {
            override fun onResponse(call: Call<List<Planta>>, response: Response<List<Planta>>) {
                if (response.isSuccessful) {
                    val plantas = response.body()
                    if (plantas != null && plantas.isNotEmpty()) {
                        plantaAdapter.updatePlantas(plantas)
                        // Debug log to check data
                        Log.d("AlmanaqueActivity", "Plantas fetched: ${plantas.size}")
                        plantas.forEach {
                            Log.d("AlmanaqueActivity", "Planta: ${it.nombre}, ${it.descripcion}")
                        }
                    } else {
                        Log.d("AlmanaqueActivity", "No plantas found")
                    }
                } else {
                    Log.e("AlmanaqueActivity", "Response not successful: ${response.message()}")
                }
            }

            override fun onFailure(call: Call<List<Planta>>, t: Throwable) {
                Log.e("AlmanaqueActivity", "Error fetching plantas: ${t.message}")
            }
        })
    }
}
